dot/sync/chain_processor_test.go

```diff
@@ -0,0 +1,607 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/runtime/mocks"
+	"github.com/ChainSafe/gossamer/lib/runtime/storage"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_chainProcessor_handleBlock_baseCase(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+
+	block := &types.Block{
+		Header: types.Header{
+			Number: 0,
+		},
+		Body: types.Body{},
+	}
+
+	mockBlockState := NewMockBlockState(ctrl)
+	mockHeader := &types.Header{
+		Number:    0,
+		StateRoot: trie.EmptyHash,
+	}
+	mockHeaderHash := mockHeader.Hash()
+	mockBlock := &types.Block{
+		Header: types.Header{
+			Number: 0,
+		},
+		Body: types.Body{},
+	}
```

Comment:
nit that would work the same, imo let's leave zero-valued fields undeclared. Same for variable declarations around.

https://github.com/uber-go/guide/blob/master/style.md#omit-zero-value-fields-in-structs

```suggestion
	mockBlock := &types.Block{}
```